libpod/container_internal_linux.go

```diff
@@ -416,9 +417,25 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 
 	// Look up and add groups the user belongs to, if a group wasn't directly specified
 	if !strings.Contains(c.config.User, ":") {
-		for _, gid := range execUser.Sgids {
-			// FIXME: We need to add a flag to containers.conf to not add these for HPC Users.
-			g.AddProcessAdditionalGid(uint32(gid))
+		var haveEnoughGids bool
+
+		if !rootless.IsRootless() {
+			// Running as root, assume there are enough GIDs.
+			haveEnoughGids = true
```

Comment:
This seems to be still error prone.  I can run a root container with a user namespace with only 100 GIDs?
podman run --usermap=0:100000:100 ...